::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
: Copyright (c) 2017 Enea AB and others.
:
: All rights reserved. This program and the accompanying materials
: are made available under the terms of the Apache License, Version 2.0
: which accompanies this distribution, and is available at
: http://www.apache.org/licenses/LICENSE-2.0
::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
From: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
Date: Tue, 18 Jul 2017 19:17:24 +0200
Subject: [PATCH] network: public: Use arm-virtual2 POD config

grep -e "10\.16\.0\." -R . -l | \
  xargs sed -i \
    -e 's/10\.16\.0\./10.0.9./g' \
    -e 's/10\.0\.9\.254/10.0.9.200/g' \
    -e 's/10\.0\.9\.1/10.0.9.254/g'

NOTE: This should be converted into a dynamic configuration read from
the universal POD descriptor in securedlab, once that is ready.

Until then, just align the public network configuration used by the
virtual POD with the Enea lab configuration specific to arm-virtual2
(i.e. public network on 10.0.9.0/24).

NOTE: Replace the gateway at 10.16.0.1 (now 10.0.9.1) with the same
IP address as our lab's gateway (10.0.9.254), to keep both possible
network layouts in sync (using all virtual networks created via
virsh, respectively our lab's static config).
This will ensure deploys continue to work in both enviroments.

Also, since our new gateway resides at 10.0.9.254, trim the DHCP pool
range for the public network to not include that address.

Signed-off-by: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
---
 mcp/config/states/networks                                          | 6 +++---
 .../classes/cluster/virtual-mcp-pike-common-noha/infra/config.yml   | 6 +++---
 .../classes/cluster/virtual-mcp-pike-common-noha/openstack_init.yml | 2 +-
 mcp/scripts/net_public.xml                                          | 2 +-
 4 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/mcp/config/states/networks b/mcp/config/states/networks
index d4735ad..e566903 100755
--- a/mcp/config/states/networks
+++ b/mcp/config/states/networks
@@ -18,7 +18,7 @@ PUBLIC_NET=$(salt --out yaml 'cmp*' pillar.get _param:external_address | \
   pillar.get _param:openstack_compute_node01_external_address | \
   awk --re-interval '/[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}/{print $2; exit}')

-[ -n "${PUBLIC_NET}" ] && PUBLIC_NET="${PUBLIC_NET%.*}.0/24" || PUBLIC_NET="10.16.0.0/24"
+[ -n "${PUBLIC_NET}" ] && PUBLIC_NET="${PUBLIC_NET%.*}.0/24" || PUBLIC_NET="10.0.9.0/24"

 salt 'ctl01*' cmd.run ". /root/keystonercv3; \
   openstack compute service list; \
@@ -29,6 +29,6 @@ salt 'ctl01*' cmd.run ". /root/keystonercv3; \
   openstack network create --external --default --provider-network-type flat \
     --provider-physical-network physnet1 floating_net"
 salt 'ctl01*' cmd.run ". /root/keystonercv3; \
-  openstack subnet create --gateway ${PUBLIC_NET%.*}.1 --no-dhcp \
-    --allocation-pool start=${PUBLIC_NET%.*}.130,end=${PUBLIC_NET%.*}.254 \
+  openstack subnet create --gateway ${PUBLIC_NET%.*}.254 --no-dhcp \
+    --allocation-pool start=${PUBLIC_NET%.*}.130,end=${PUBLIC_NET%.*}.200 \
     --network floating_net --subnet-range ${PUBLIC_NET} floating_subnet"
diff --git a/mcp/reclass/classes/cluster/virtual-mcp-pike-common-noha/infra/config.yml b/mcp/reclass/classes/cluster/virtual-mcp-pike-common-noha/infra/config.yml
index 3c3d526..e114ec2 100644
--- a/mcp/reclass/classes/cluster/virtual-mcp-pike-common-noha/infra/config.yml
+++ b/mcp/reclass/classes/cluster/virtual-mcp-pike-common-noha/infra/config.yml
@@ -61,13 +61,13 @@ parameters:
           params:
             single_address: 172.16.10.105
             tenant_address: 10.1.0.105
-            external_address: 10.16.0.105
+            external_address: 10.0.9.105
         openstack_compute_node02:
           params:
             single_address: 172.16.10.106
             tenant_address: 10.1.0.106
-            external_address: 10.16.0.106
+            external_address: 10.0.9.106
         openstack_gateway_node01:
           params:
             tenant_address: 10.1.0.110
-            external_address: 10.16.0.110
+            external_address: 10.0.9.110
diff --git a/mcp/reclass/classes/cluster/virtual-mcp-pike-common-noha/openstack_init.yml b/mcp/reclass/classes/cluster/virtual-mcp-pike-common-noha/openstack_init.yml
index cf35a2b..d677420 100644
--- a/mcp/reclass/classes/cluster/virtual-mcp-pike-common-noha/openstack_init.yml
+++ b/mcp/reclass/classes/cluster/virtual-mcp-pike-common-noha/openstack_init.yml
@@ -14,7 +14,7 @@ parameters:
     openstack_region: RegionOne
     admin_email: root@localhost
     cluster_public_protocol: http
-    cluster_public_host: 10.16.0.101
+    cluster_public_host: 10.0.9.101
     neutron_public_protocol: http
     neutron_control_dvr: 'False'
     neutron_l3_ha: 'False'
diff --git a/mcp/scripts/net_public.xml b/mcp/scripts/net_public.xml
index d6df4aa..87f3146 100644
--- a/mcp/scripts/net_public.xml
+++ b/mcp/scripts/net_public.xml
@@ -10,5 +10,5 @@
   <name>public</name>
   <bridge name="public"/>
   <forward mode="nat"/>
-  <ip address="10.16.0.1" netmask="255.255.255.0" />
+  <ip address="10.0.9.254" netmask="255.255.255.0" />
 </network>
